x/superfluid/keeper/msg_server_test.go

```diff
@@ -597,3 +598,98 @@ func (s *KeeperTestSuite) TestAddToConcentratedLiquiditySuperfluidPosition_Event
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestSetDenomRiskFactors() {
+	s.SetupTest()
+
+	msgServer := keeper.NewMsgServerImpl(s.App.SuperfluidKeeper)
+	c := sdk.WrapSDKContext(s.Ctx)
+
+	govAddr := s.App.AccountKeeper.GetModuleAccount(s.Ctx, govtypes.ModuleName).GetAddress().String()
+
+	// Let's turn this into a table test
+	setDenomRiskFactorTests := []struct {
+		name          string
+		sender        string
+		denom         string
+		riskFactor    string
+		expectedError string
+	}{
+		{"happy path", govAddr, "something", "0.1", ""},
+		{"happy path with zero", govAddr, "something", "0", ""},
+		{"bad gov addr", "osmo1herasn5ewvv9acpujdmqxz698y849aq9ucsccl", "something", "0", "only the governance module is allowed to execute this message"},
+		{"bad gov addr 2", "something else", "something", "0", "invalid sender address (decoding bech32 failed"},
+		{"bad denom", govAddr, "", "0", "denom cannot be empty"},
+		{"bad risk factor", govAddr, "something", "NaN", "invalid risk factor (failed to set decimal string"},
```

Comment:
nit: Since this is a new test, can this be refactored to show the struct name next to each entry? It makes it 100x easier to parse rather than matching each field up to its respective struct field.

Also, can the expectedError be of type error, so we can use error types instead of hard coding the text of the error?

Reply:
I like using strings for the error because that way I can do partial matching (`Contains`). Other than that, yeah, we can refactor